Start preparation for TIM2025 workshop (#925)
computational-cell-analytics/micro-sam: workshops/tim_2025/README.md

# TIM 2025 Workshop: Segment Anything for Microscopy

This document walks you through the preparation for the upcoming TIM 2025 workshop on "Segment Anything for Microscopy".

## Workshop Overview

The workshop will be three hours and will be divided into three parts:
1. Step 0: Install `micro-sam` on HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).
2. Short introduction (15 minutes, you can find the slides [here](https://docs.google.com/presentation/d/1jEmjHLOCMefzU8W3hBMGh7ooseEIfoe-/edit?usp=sharing&ouid=113044948772353505255&rtpof=true&sd=true))
3. Using the `micro_sam` napari plugin (on HIVE) for interactive (and automatic) 2d and 3d segmentation (30 minutes).
4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model or an advanced application (60 minutes).
5. Discussion on "What you can do next" (30 minutes)

We will walk through how to use the `micro_sam` plugin for interactive segmentation in part 3, so that you can then apply it to your own data (or the example data that is most similar to your targeted application) in part 4.

> Alternatively, you can also work on model finetuning or an advanced application, such as using the `micro_sam` in part 4.

Reviewer: ```suggestion
Alternatively, you can also work on model finetuning or an advanced application, such as using the `micro_sam` pythhon library in part 4.
```

And finally, we have a discussion on your experience and feedback / possible improvements in part 5. We will also try to formulate a segmentation workflow applicable for your data "at home".

**Please read the [Workshop Preparation](#workshop-preparation) section carefully and follow the relevant steps to get started.**

## Workshop Preparation

To get started for the workshop, please do the following:
- Install the latest version of `micro_sam`, see [Installation](#installation) for details

> - Download the models and pre-computed embeddings for the common 3D segmentation, see [here](#download-embeddings-for-3d-segmentation).

Reviewer: ```suggestion
- Download the models and pre-computed embeddings for the 3D segmentation problems, see [here](#download-embeddings-for-3d-segmentation).
```

    - High-throughput annotation of cells (or other structures) in 2D images, see [high-throughput-annotation](#high-throughput-image-annotation).
    - 3D segmentation in light or electron microscopy, see [3D LM segmentation](#3d-lm-segmentation) and [3D EM segmentation](#3d-em-segmentation).
    - Finetuning SAM on custom data, see [model finetuning](#model-finetuning).

You can do all of this on your laptop with a CPU, except for model finetuning, which requires a GPU. However, we have GPU resources available on HIVE. And we have prepared a notebook that runs on cloud resources with a GPU for this.

If you want to learn more about the `micro_sam` napari plugin or python library you can check out the [documentation](https://computational-cell-analytics.github.io/micro-sam/) and our [tutorial videos](https://youtube.com/playlist?list=PLwYZXQJ3f36GQPpKCrSbHjGiH39X4XjSO&si=3q-cIRD6KuoZFmAM).

### Installation

Please make sure to install the latest version of `micro_sam` before the workshop using `conda` (or `mamba`).
You can create a new environment and install it like this:

```bash
conda create -n micro_sam -c conda-forge python=3.11 natsort
conda activate micro_sam

> conda install -c conda-forge "micro_sam>=1.3.1" "pytorch>=2.5" "protobuf<5" "nifty=1.2.1=*_4"

Reviewer: We need to update these instructions so that it matches what I announced in the mattermost. I will follow up on this.

Author: Okie.

```

If you already have an installation of `micro_sam` please update it by running the last command in your respective environment. You can find more information about the installation [here](https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#installation).

<details>
    <summary>HIVE Installation Instructions</summary>
    To install "micro-sam" on HIVE, please follow the instructions below:
    <ol>
        <li>Open PowerShell</li>
        <li>Run `E:\TiM2025-Software\Miniforge3-Windows-x86_64.exe` in your termnial to install MiniForge. It will guide you through all instructions. Please follow along to install MiniForge successfully.</li>
        <li>Once the installation is done, you need to initialize your conda environments using `.\miniforge3\Scripts\conda.exe init powershell`.</li>
        <li>And here onwards, follow the installation instructions above.</li>
    </ol>
</details>

### Download Embeddings for 3D EM Segmentation

> We provide a script to download the models used in the workshop. To run the script you first need to use `git` to download this repository:

Reviewer: ```suggestion
We provide a script to download the models used in the workshop. To run the script you first need to download this repository. If you haven't done this already, then please use `git` to download it:
```

```bash

> git clone https://github.com/computational-cell-analytics/micro-sam

Reviewer: Need to checkout dev

Author: Ahha yes you are right. I am testing this on HIVE now, and will add once your review is done.

You can use the [Image Series Annotator](https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#image-series-annotator) to run interactive segmentation for many images stored in a folder.
This annotation mode is well suited for generating annotations for 2D cell segmentation or similar analysis tasks.

We have prepared an example dataset for the workshop that you can use. It consists of 15 images from the [CellPose](https://www.cellpose.org/) dataset. You can download the data with the script `download_dataset.py`:

```bash
python download_datasets.py -i data -d cells
```

This will download the data to the folder `data/cells` with images stored in the subfolder `images` and segmentation masks in `masks`. After this you can start the image series annotation tool, either via the napari plugin (we will show this in the workshop) or via the command line:

```bash
micro_sam.image_series_annotator -i data/cells/images -o annotations/cells -e embeddings/cells/vit_b_lm -m vit_b_lm
```

Note: You can use `micro_sam` with different models: the original models from Segment Anything and models finetuned for different microscopy segmentation tasks by us.

> For cell segmentation you can either use `vit_b` (the original model) or `vit_b_lm` (our model). Our `vit_b_lm` model will be better for most cell segmentation tasks but there may be cases where `vit_b` is better, so it makes sense to test both before annotating your data. Please refer to [our documentation](https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#finetuned-models) for details on the models.

Reviewer: ```suggestion
For cell segmentation you can either use `vit_b` (the original SAM) or `vit_b_lm` (our model; the default model used by `micro_sam`). Our `vit_b_lm` model will be better for most cell segmentation tasks but there may be cases where `vit_b` is better, so it makes sense to test both before annotating your data. Please refer to [our documentation](https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#finetuned-models) for details on the models.
```

# TIM 2025 Workshop: Segment Anything for Microscopy

This document walks you through the preparation for the upcoming TIM 2025 workshop on "Segment Anything for Microscopy".

## Workshop Overview

The workshop will be three hours and will be divided into three parts:
1. Step 0: Install `micro-sam` on HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).

> 2. Short introduction (30 minutes, you can find the slides [here](TODO))

Reviewer: 30 minutes is too long for this.

3. Using the `micro_sam` napari plugin (on HIVE) for interactive (and automatic) 2d and 3d segmentation (30 minutes).
4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model or an advanced application (60 minutes).
5. Discussion on "What you can do next" (30 minutes)

We will walk through how to use the `micro_sam` plugin for interactive segmentation in part 3, so that you can then apply it to your own data (or the example data that is most similar to your targeted application) in part 4.
Alternatively, you can also work on model finetuning or an advanced application, such as using the `micro_sam` in part 4.

> And finally, we have an open discussion on your experience and expectations from `micro_sam` in part 5, and the take home segmentation workflow for you all.

Reviewer: ```suggestion
And finally, we have a discussion on your experience and feedback / possible improvements in part 5. We will also try to formulate a segmentation workflow applicable for your data "at home".
```

The workshop will be three hours and will be divided into three parts:

> 1. Step 0: Install `micro-sam` on HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).

Reviewer: ```suggestion
1. Step 0: Install `micro-sam` on your laptop or HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).
```

2. Short introduction (15 minutes, you can find the slides [here](https://docs.google.com/presentation/d/1jEmjHLOCMefzU8W3hBMGh7ooseEIfoe-/edit?usp=sharing&ouid=113044948772353505255&rtpof=true&sd=true))

> 3. Using the `micro_sam` napari plugin (on HIVE) for interactive (and automatic) 2d and 3d segmentation (30 minutes).

Reviewer: ```suggestion
3. Using the `micro_sam` napari plugin for interactive (and automatic) 2d and 3d segmentation (30 minutes).
```

> 4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model or an advanced application (60 minutes).

Reviewer: ```suggestion
4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model , or following an advanced example application (60 minutes).
```
